### PR TITLE
feature/CP-4719-device-restored-from-icloud-backup-is-not-subscribed

### PR DESCRIPTION
**Pull request by cpandya25**
CP-4719-device-token updated.

### CODE REVIEW

File: CleverPush/Source/CleverPushInstance.m [hunk 1]
```diff
@@ -1081,6 +1081,15 @@ - (void)handleDidFailRegisterForRemoteNotification:(NSError*)err {
 
 #pragma mark - register Device Token
 - (void)registerDeviceToken:(id)newDeviceToken onSuccess:(CPResultSuccessBlock)successBlock onFailure:(CPFailureBlock)failureBlock {
+    if([[NSUserDefaults standardUserDefaults] objectForKey:CLEVERPUSH_DEVICE_TOKEN_KEY] != nil) {
```
Comment (gbhrdt) [inline CleverPush/Source/CleverPushInstance.m:1084]: Code style: space after `if ` missing
2 if's can be combined to one if

Comment (gbhrdt) [reply CleverPush/Source/CleverPushInstance.m:1084]: Also we can simplify by puting the conditions below here:
https://github.com/cleverpush/cleverpush-ios-sdk/pull/88/files#diff-f5326f84792597a7145e9b2f5be8823e3bc651f5b4faac02dc05dca951eee90fR1121

Comment (gbhrdt) [reply CleverPush/Source/CleverPushInstance.m:1084]: Because of the early-exits, at the bottom of this function we are sure that new != old device token.

**Review by gbhrdt — CHANGES_REQUESTED** [review]

File: CleverPush/Source/CleverPushInstance.m (hunk 1 above)
Comment (cpandya25) [reply CleverPush/Source/CleverPushInstance.m:1084]: > Also we can simplify by puting the conditions below here: https://github.com/cleverpush/cleverpush-ios-sdk/pull/88/files#diff-f5326f84792597a7145e9b2f5be8823e3bc651f5b4faac02dc05dca951eee90fR1121

Hello @gbhrdt Can you please give the correct link?

Comment (gbhrdt) [reply CleverPush/Source/CleverPushInstance.m:1084]: @cpandya25  Line 1121

image

**Review by gbhrdt — CHANGES_REQUESTED** [review]

File: CleverPush/Source/CleverPushInstance.m (hunk 1 above)
Comment (cpandya25) [reply CleverPush/Source/CleverPushInstance.m:1084]: @gbhrdt For the confirmation, I am asking that we need to put this code.

Comment (gbhrdt) [reply CleverPush/Source/CleverPushInstance.m:1084]: Yes, but then we probably dont need the condition. Please check the code flow.

Comment (cpandya25) [reply CleverPush/Source/CleverPushInstance.m:1084]: @gbhrdt Okay, let me check how the token changes work another time with restoring the device.

Comment (cpandya25) [reply CleverPush/Source/CleverPushInstance.m:1084]: Before taking backup 
Screenshot 2022-10-18 at 10 54 03 AM

Comment (cpandya25) [reply CleverPush/Source/CleverPushInstance.m:1084]: After taking backup
Screenshot 2022-10-18 at 11 08 31 AM

**Review by gbhrdt — APPROVED** [review]

**Pull request merged**